File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/OuterScheduleOfInstructor.kt
package com.example.drivingschoolappandroidclient.api.models

class OuterScheduleOfInstructor(
    googleSheetId: String,
    googleSheetPageName: String,
    timesOfClassesRange: String,
    datesOfClassesRange: String,
    classesRange: String,
    freeClassExampleRange: String,
    notFreeClassExampleRange: String?,
    yearRange: String

) : OuterScheduleOfInstructorModel(
    googleSheetId,
    googleSheetPageName,
    timesOfClassesRange,
    datesOfClassesRange,
    classesRange,
    freeClassExampleRange,
    notFreeClassExampleRange,
    yearRange
){
    /**
     * Id внешнего расписания
     */
    var outerScheduleId : Int? = null

    /**
     * Id инструктора, которому принадлежит это внешнее расписание
     */
    var instructorId : Int? = null
    /**
     * Инструктор, которому принадлежит это внешнее расписание
     */
    var instructor : Instructor? = null

    companion object {
        fun constructor(model : OuterScheduleOfInstructorModel) = OuterScheduleOfInstructor(
            googleSheetId = model.googleSheetId,
            googleSheetPageName = model.googleSheetPageName,
            timesOfClassesRange = model.timesOfClassesRange,
            datesOfClassesRange = model.datesOfClassesRange,
            classesRange = model.classesRange,
            freeClassExampleRange = model.freeClassExampleRange,
            notFreeClassExampleRange = model.notFreeClassExampleRange,
            yearRange = model.yearRange
        )
    }
    // TODO: добавить все методы по получению данных, добавлению и проверке внутренних расписаний сюда
}

File: PDP/DrivingSchoolWebAPI3/DrivingSchoolAPIModels/DbModels/OuterScheduleOfInstructor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DrivingSchoolAPIModels
{
    public class OuterScheduleOfInstructor : OuterScheduleOfInstructorModel
    {
        /// <summary>
        /// Id внешнего расписания
        /// </summary>
        [Key]
        public int OuterScheduleId { get; set; }

        /// <summary>
        /// Id инструктора, которому принадлежит это внешнее расписание
        /// </summary>
        public int InstructorId { get; set; }
        /// <summary>
        /// Инструктор, которому принадлежит это внешнее расписание
        /// </summary>
        [ForeignKey(nameof(InstructorId))]
        public Instructor Instructor { get; set; }

        //public OuterScheduleOfInstructor(OuterScheduleOfInstructorModel model)
        //{
        //    InstructorId = model.InstructorId;
        //    GoogleSheetId = model.GoogleSheetId;
        //    GoogleSheetPageName = model.GoogleSheetPageName;
        //    TimesOfClassesRange = model.TimesOfClassesRange;
        //    DatesOfClassesRange = model.DatesOfClassesRange;
        //    ClassesRange = model.ClassesRange;
        //    FreeClassExampleRange = model.FreeClassExampleRange;
        //    NotFreeClassExampleRange = model.NotFreeClassExampleRange;
        //    YearRange = model.YearRange;
        //}
        public static OuterScheduleOfInstructor Constructor(OuterScheduleOfInstructorModel model)
        {
            return new OuterScheduleOfInstructor
            {
                GoogleSheetId = model.GoogleSheetId,
                GoogleSheetPageName = model.GoogleSheetPageName,
                TimesOfClassesRange = model.TimesOfClassesRange,
                DatesOfClassesRange = model.DatesOfClassesRange,
                ClassesRange = model.ClassesRange,
                FreeClassExampleRange = model.FreeClassExampleRange,
                NotFreeClassExampleRange = model.NotFreeClassExampleRange,
                YearRange = model.YearRange
            };
        }
        // TODO: добавить все методы по получению данных, добавлению и проверке внутренних расписаний сюда
    }
}
